### DISCUSSION TITLE
How to automatically crop similar images to the same maximal section

### DISCUSSION THREAD

**Discussion by halloleo** [ImageMagick/ImageMagick#2715] [General]
I have some hand-held photos with a moving person in front of the same background. So teh camera does not move with the person but stays on teh background - howver the photos are hand-held, so there are _some_ variations in the background area the photos show.. I.e. the images do not align perfectly.

I now want to crop all images automatically as if the images were transparent sldies and I' do teh following: I layered them on top of each other, so that the backgrounds always match up (as good as possible).  Then I crop all images to the larges rectangle all images share.

Can I do this with ImageMagick? Or are there other solutions more appropriate?

**Comment by snibgo** [comment]
I'm not clear what you want.

Do you want to align the photos? If so, then if the camera hasn't moved then a "-distort perspective" can be applied to each image other than a master image, to make them all align with the master. To do that, you need to do searches to find corresponding features in the images.

Perhaps the Hugin toolset can do that more directly.

I don't know what you mean by "maximal section".

**Reply by halloleo** [reply]
Sorry, if my question wasn't clear: 

Figuratively speaking I want layer the photos on top of each other, so the the backgrounds always lay on top of each other (as good as possible) and _then_ I crop all images to the larges rectangle all images share. That's what I meant with "maximal section".

I have updated teh question accordingly.

**Comment by snibgo** [comment]
On the alignment, see my pages [Simple alignment by matching areas](http://im.snibgo.com/simpalign.htm) and [Alignment by matching points](http://im.snibgo.com/alignpnt.htm) and others. This will give a required "+distort perspective" for each image to align with one that you choose to be the master.

On the "maximal section", do a "+distort perspective" on white images with black virtual pixels. Composite them with "-compose darken". The result is white where every image has a pixel, and otherwise black. Trim off all the black pixels. The page geometry of the result is the maximal section.

**Reply by halloleo** [reply]
Thanks @snibgo! Had a look and your scripts look pretty cool!

However by further researching the topic I found that Fiji/ImageJ with its "Register Virtual Stack Slices" plugin in my case might be an easier solution.
